### ISSUE TITLE
Undesired bounding sphere calculation and exception with custom buffer layout and custom shader

### ISSUE THREAD

**Issue by vagran** [mrdoob/three.js#27823]
### Description

I use three.js to draw 2D graphics. So I have `BufferGeometry` with custom layout (2D coordinates only) and `RawShaderMaterial` to handle this data in scope of `Mesh` object. It worked flawlessly until some recent update when this exception started to appear:
```
THREE.BufferGeometry.computeBoundingSphere(): Computed radius is NaN. The "position" attribute is likely to have NaN values.
```
The bounding sphere calculation is called implicitly by `rener()` method:
```
computeBoundingSphere	@ 
intersectsObject	@
projectObject	@
projectObject	@
WebGLRenderer.render
```
Looks like some object cooling logic. Is there any way to disable it? Or maybe I miss some hint in the code to indicate non-conventional geometry buffer layout?

### Reproduction steps

1. Run the sample code

### Code

```js
import { BufferGeometry, Float32BufferAttribute, Mesh, RawShaderMaterial, 
        PerspectiveCamera, Scene, WebGLRenderer, GLSL3 } from 'three';

// Create a scene
const scene = new Scene();

// Create a camera
const camera = new PerspectiveCamera(75, window.innerWidth / window.innerHeight, 0.1, 1000);
camera.position.z = 3;

// Create a renderer
const renderer = new WebGLRenderer();
renderer.setSize(window.innerWidth, window.innerHeight);
document.body.appendChild(renderer.domElement);

// Create a BufferGeometry object
const geometry = new BufferGeometry();

// Define the vertices for the square
const vertices = [
    -1.0, -1.0,
     1.0, -1.0,
     1.0,  1.0,
    -1.0,  1.0
];

// Define the faces of the square
const indices = [
    0, 1, 2, 
    0, 2, 3
];

// Add the vertices and faces to the geometry
geometry.setAttribute('position', new Float32BufferAttribute(vertices, 2));
geometry.setIndex(indices);

// Create a ShaderMaterial
const material = new RawShaderMaterial({
    vertexShader: `
        in vec2 position;
        uniform mat4 modelViewMatrix;
        uniform mat4 projectionMatrix;
        void main() {
            gl_Position = projectionMatrix * modelViewMatrix * vec4(position, 0.0, 1.0);
        }
    `,
    fragmentShader: `
        precision highp float;
        precision highp int;
        out vec4 fragColor;
        void main() {
            fragColor = vec4(0.0, 1.0, 0.0, 1.0); // green color
        }
    `,
    depthTest: false,
    depthWrite: false,
    glslVersion: GLSL3
});

// Create a mesh with the geometry and material
const cube = new Mesh(geometry, material);

// Add the cube to the scene
scene.add(cube);

renderer.render(scene, camera);
```

### Live example

* [codesandbox.io](https://codesandbox.io/p/sandbox/eager-sun-pxgfr6?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clt2qm9o80006396h9lzyzkj0%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clt2qm9o70002396h6nczv0iz%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clt2qm9o70003396hffz0ssuu%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clt2qm9o70005396h5kq7zw2g%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clt2qm9o70002396h6nczv0iz%2522%253A%257B%2522id%2522%253A%2522clt2qm9o70002396h6nczv0iz%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clt2r63sj0002396g3x95h4pc%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A54%252C%2522startColumn%2522%253A10%252C%2522endLineNumber%2522%253A54%252C%2522endColumn%2522%253A10%257D%255D%252C%2522filepath%2522%253A%2522%252Fsrc%252Findex.js%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clt2r63sj0002396g3x95h4pc%2522%257D%252C%2522clt2qm9o70005396h5kq7zw2g%2522%253A%257B%2522id%2522%253A%2522clt2qm9o70005396h5kq7zw2g%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clt2qm9o70004396hf3h7ekyu%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clt2qm9o70004396hf3h7ekyu%2522%257D%252C%2522clt2qm9o70003396hffz0ssuu%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clt2qm9o70003396hffz0ssuu%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

### Version

r161.0

### Device

Desktop

### Browser

Chrome

**Comment by Mugen87** [comment]
Try this:
```js
cube.frustumCulled = false;
```
>It worked flawlessly until some recent update when this exception started to appear:

That's strange since view frustum culling was always enabled for meshes with custom shaders. Note that we do not officially support 2D geometry data so this issue won't be fixed.

**Comment by vagran** [comment]
> Try this:
> 
> ```js
> cube.frustumCulled = false;
> ```
I tried, it has the same error. [Example](https://codesandbox.io/p/sandbox/dark-forest-pxgfr6?file=%2Fsrc%2Findex.js%3A63%2C1&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clt2qm9o80006396h9lzyzkj0%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clt2qm9o70002396h6nczv0iz%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clt2qm9o70003396hffz0ssuu%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clt2qm9o70005396h5kq7zw2g%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clt2qm9o70002396h6nczv0iz%2522%253A%257B%2522id%2522%253A%2522clt2qm9o70002396h6nczv0iz%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clt2r63sj0002396g3x95h4pc%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252Findex.js%2522%252C%2522state%2522%253A%2522IDLE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A63%252C%2522startColumn%2522%253A1%252C%2522endLineNumber%2522%253A63%252C%2522endColumn%2522%253A1%257D%255D%257D%255D%252C%2522activeTabId%2522%253A%2522clt2r63sj0002396g3x95h4pc%2522%257D%252C%2522clt2qm9o70005396h5kq7zw2g%2522%253A%257B%2522id%2522%253A%2522clt2qm9o70005396h5kq7zw2g%2522%252C%2522activeTabId%2522%253A%2522clt2qm9o70004396hf3h7ekyu%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clt2qm9o70004396hf3h7ekyu%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522path%2522%253A%2522%252F%2522%257D%252C%257B%2522type%2522%253A%2522SANDBOX_INFO%2522%252C%2522id%2522%253A%2522clt2sfscx001o396ge2om7sqa%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%255D%257D%252C%2522clt2qm9o70003396hffz0ssuu%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clt2qm9o70003396hffz0ssuu%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D).

**Comment by vagran** [comment]
> That's strange since view frustum culling was always enabled for meshes with custom shaders. Note that we do not officially support 2D geometry data so this issue won't be fixed.

@Mugen87 I re-checked my production code and figured out that I already had `.frustumCulled = false`. So seems it was working some time ago, and now is broken. Could we re-open this issue?

**Comment by Mugen87** [comment]
The documentation is wrong at this point. It has been corrected here: #27744

**Comment by vagran** [comment]
Thanks for the explanation and advice. I think this unconditional calculation should be stated in the documentation. The documentation fix above states "The engine automatically computes the bounding sphere when it is needed, e.g., for ray casting or view frustum culling." however it is actually _always_ computed now.
